### ISSUE TITLE
Gemfile order matters

### ISSUE THREAD

**Issue by jclusso** [amazing-print/amazing_print#107]
If you use this gem and have it placed before Mongoid in your Gemfile, it doesn't work. Is there a way to fix this gem so where it gets placed in the Gemfile doesn't matter?

**Comment by danarnold** [comment]
When testing locally, I was able to fix this by adding a `require mongoid` before `defined?(Mongoid)` is checked in lib/amazing_print.rb:

```ruby
  begin
    require 'mongoid'
  rescue LoadError
  end
  require_relative 'amazing_print/ext/mongoid'        if defined?(Mongoid)
```

I assume the same could be done for all of the other `defined?` checks as well in order to prevent Gemfile order load issues. I don't know if there's a more canonical way of doing this, and I don't love this solution. But I figured it makes sense that if the problem is having something in the `$LOAD_PATH` that just hasn't been required yet, then requiring it would be an easy fix.
